go/libraries/doltcore/doltdb/root_val.go

```diff
@@ -135,6 +137,234 @@ func (root *RootValue) HasTable(ctx context.Context, tName string) (bool, error)
+	m, err := root.getOrCreateSuperSchemaMap(ctx)
+
+	if err != nil {
+		return false, "", err
+	}
+
+	err = m.Iter(ctx, func(key, value types.Value) (stop bool, err error) {
+		ssValRef := value.(types.Ref)
+		ssVal, err := ssValRef.TargetValue(ctx, root.vrw)
+
+		if err != nil {
+			return true, err
+		}
+
+		ss, err := encoding.UnmarshalSuperSchemaNomsValue(ctx, root.vrw.Format(), ssVal)
+
+		if err != nil {
+			return true, err
+		}
+
+		_, found = ss.GetColumn(tag)
+
+		if found {
+			tblName = key.(types.String).HumanReadableString()
+		}
+
+		return found, nil
+	})
+
+	if err != nil {
+		return false, "", err
+	}
+
+	if found {
+		return found, tblName, err
+	}
+
+	// Now check if the tag exists in the working set of any table
+	tNames, err := root.GetTableNames(ctx)
+
+	if err != nil {
+		return false, "", err
+	}
+
+	for _, tn := range tNames {
+		t, _, err := root.GetTable(ctx, tn)
+
+		if err != nil {
+			return false, "", err
+		}
+
+		sch, err := t.GetSchema(ctx)
+
+		if err != nil {
+			return false, "", err
+		}
+
+		_, found = sch.GetAllCols().GetByTag(tag)
+
+		if found {
+			return found, tn, nil
+		}
+	}
+
+	return false, "", nil
+}
+
+// GetSuperSchema returns the SuperSchema for the table name specified if that table exists.
+func (root *RootValue) GetSuperSchema(ctx context.Context, tName string) (*schema.SuperSchema, bool, error) {
+	// SuperSchema is only persisted on Commit()
+	ss, found, err := root.getStaleSuperSchema(ctx, tName)
+
+	if err != nil {
+		return nil, false, err
+	}
+	if !found {
+		ss, _ = schema.NewSuperSchema()
+	}
+
+	t, tblFound, err := root.GetTable(ctx, tName)
+
+	if err != nil {
+		return nil, false, err
+	}
+
+	if !found && !tblFound {
+		// table doesn't exist in current commit or in history
+		return nil, false, nil
+	}
+
+	if tblFound {
+		sch, err := t.GetSchema(ctx)
+
+		if err != nil {
+			return nil, false, err
+		}
+
+		err = ss.AddSchemas(sch)
+
+		if err != nil {
+			return nil, false, err
+		}
+	}
+
+	return ss, true, err
+}
+
+// GetUniqueTagFromNomsKinds returns a tag that has not yet been used in the history of this root.
+func (root *RootValue) GetUniqueTagFromNomsKinds(ctx context.Context, schKinds []types.NomsKind) (uint64, error) {
+	ssMap, err := root.getOrCreateSuperSchemaMap(ctx)
+
+	if err != nil {
+		return schema.InvalidTag, err
+	}
+
+	var sss []*schema.SuperSchema
+	err = ssMap.Iter(ctx, func(key, value types.Value) (stop bool, err error) {
+		ssValRef := value.(types.Ref)
+		ssVal, err := ssValRef.TargetValue(ctx, root.vrw)
+
+		if err != nil {
+			return true, err
+		}
+
+		ss, err := encoding.UnmarshalSuperSchemaNomsValue(ctx, root.vrw.Format(), ssVal)
+
+		if err != nil {
+			return true, err
+		}
+
+		sss = append(sss, ss) // go get -f parseltongue
+		return false, nil
+	})
+
+	if err != nil {
+		return schema.InvalidTag, err
+	}
+
+	rootSuperSchema, err := schema.SuperSchemaUnion(sss...)
```

Comment:
...I remember reworking this function once or twice, so I think this may have been motivated by the comment TBH. 
But yeah, they're not that big and I'm not sure it would have us anything to union them on the go